api/handlers/annotations.go

```diff
@@ -91,19 +147,110 @@ func GetAnnotations(ctx *fiber.Ctx) error {
 		return api.InvalidRequestURL(err)
 	}
 
-	// Placeholder code: returns an empty result.
-	// TODO: implement fetching from datastore.
-	result := api.Result[AnnotationResult]{
-		Results: []AnnotationResult{},
+	format := new(api.Format)
+	if err := ctx.QueryParser(format); err != nil {
+		return api.InvalidRequestURL(err)
+	}
+
+	// NOTE: fiber's QueryParser does not handle map[string]string so we need to parse the query manually.
+	// This can be revisited if PR https://github.com/gofiber/fiber/issues/2524 is merged.
+	qry.Observation = make(map[string]string)
+	for k, v := range ctx.Queries() {
```

Comment:
so you can't use the std json package Marshal and Unmarshal functionality ?

Reply:
It is not JSON. This is parsing the query parameters in the url